lib/distillery/tasks/release.ex

```diff
@@ -19,6 +19,9 @@ defmodule Mix.Tasks.Release do
       # Build an upgrade release from a specific version
       mix release --upgrade --upfrom=0.1.0
 
+      # Build an upgrade release using soft purge
```

Comment:
I would rather not document this option here, as it is really an advanced option that one should consult the documentation on using first. This keeps the output of `mix help release` more useful (generally the folks running help are using Distillery for the first time, or when they are still new to releases).